### ISSUE TITLE
SSL/TLS Handshake Latency

### ISSUE THREAD

**Issue by vanbroup** [davecheney/httpstat#113]
It would be really good to have the SSL/TLS handshake latency.

**Comment by davecheney** [comment]
@vanbroup what is missing at the moment? Please be specific.

**Comment by vanbroup** [comment]
The SSL/TLS Handshake Latency is measured from the **ClientHello** message that comes after the TCP ACK until the **Finished** message which is sent immediately after a change cipher spec message to verify that the key exchange and authentication processes were successful.

[RFC5246 Section 7.4.9 Finished](https://tools.ietf.org/html/rfc5246#section-7.4.9) states:
> The Finished message is the first one protected with the just
>       negotiated algorithms, keys, and secrets.  Recipients of Finished
>       messages MUST verify that the contents are correct.  Once a side
>       has sent its Finished message and received and validated the
>       Finished message from its peer, it may begin to send and receive
>       application data over the connection.

This allows a client to measure the delay caused by the SSL/TLS Handshake as also implemented by [httpstat.py](https://github.com/reorx/httpstat) and highlighted in this example:

![image](https://cloud.githubusercontent.com/assets/722070/19625285/8fd10946-9914-11e6-8d37-081f6228162f.png)

**Comment by juanpaulo** [comment]
@vanbroup how is it different from the current one (TLS Handshake)?

![screen shot 2016-10-25 at 2 01 45 pm](https://cloud.githubusercontent.com/assets/2783678/19673550/d0071124-9abb-11e6-9fc2-556b557d193b.png)

**Comment by vanbroup** [comment]
@juanpaulo thanks, I just tested and it's indeed already there!

I was mainly interested in the handshake part and the screenshot in the readme doesn't include the change from pull request #25, also http/httptrace doesn't expose any TLS information (until go 1.8) so I did not even try to run it honestly speaking.

Maybe it's smart to update the screenshot in README.md to relect the implementation of the SSL/TLS Handshake Latency.

**Comment by davecheney** [comment]
Since we moved to the httptrace framework we can only do things that it knows how to do. 

There is a PR pending the release of go 1.8 that may help.

**Comment by davecheney** [comment]
> I was mainly interested in the handshake part and the screenshot in the
readme doesn't include the change from pull request #25
<https://github.com/davecheney/httpstat/pull/25>,

I'm pretty sure it does, that screenshot was taken with 1.0 version of
httpstat.

If this is incorrect, it is a bug and should be fixed.

**Comment by moorereason** [comment]
@davecheney, I think his point was that the screenshot should be of a TLS request.

**Comment by davecheney** [comment]
Oh, that's my mistake. When I made the announcement I used the github page for this project.

If someone wants to update the screenshot with something more exciting, please send a PR.

**Comment by juanpaulo** [comment]
@davecheney I hope #115 is exciting enough. Had to use a different one since dave.cheney.net doesn't seem to have an SSL endpoint.

**Comment by davecheney** [comment]
That'll do, I was just using my own site as a demo, anything that offers
https will do.

**Comment by ch3lo** [comment]
nice, so I think this issue should be marked as resolved right?

**Comment by moorereason** [comment]
No, we're waiting on Go 1.8 release to get [this enhancement](https://beta.golang.org/doc/go1.8#net_http_httptrace).

**Comment by janisz** [comment]
Ping. Go was released.

**Comment by cristaloleg** [comment]
Can be closed?
